packages/worker/lib/integration.service.ts

```diff
@@ -120,9 +139,21 @@ class IntegrationService implements IntegrationServiceInterface {
                 });
             }
 
+            delete this.runningScripts[syncId];
+
             return { success: false, error: new NangoError(content, 500), response: null };
         }
     }
+
+    private sendHeartbeat() {
+        setInterval(() => {
+            Object.keys(this.runningScripts).forEach((syncId) => {
+                const context = this.runningScripts[syncId];
+
+                context?.heartbeat();
+            });
+        }, 300000);
```

Comment:
I would heartbeat more often than the timeout. For instance 10 mins heartbeat for a 30 mins timeout to deal with any transient error that would prevent temporal to receive one or two heartbeat (network glitch, temporal internal issue, etc....)

Reply:
Yup, agreed. I set this to be every 5 minutes right after I asked for review actually